exporter/internal/queue/default_batcher.go

```diff
@@ -124,25 +124,19 @@ func (qb *DefaultBatcher) startTimeBasedFlushingGoroutine() {
 			case <-qb.shutdownCh:
 				return
 			case <-qb.timer.C:
-				qb.currentBatchMu.Lock()
-				if qb.currentBatch == nil || qb.currentBatch.req == nil {
-					qb.currentBatchMu.Unlock()
-					continue
-				}
-				batchToFlush := *qb.currentBatch
-				qb.currentBatch = nil
-				qb.currentBatchMu.Unlock()
-
-				// flushAsync() blocks until successfully started a goroutine for flushing.
-				qb.flushAsync(batchToFlush)
-				qb.resetTimer()
```

Comment:
Not sure I understand this, can we do this in a separate PR?

Reply:
Thanks! Here it is: https://github.com/open-telemetry/opentelemetry-collector/pull/11666
`batch_sender_test` helped me detect that the original implementation is missing a flush on shutdown.